### PR TITLE
Assets Rules Improvment

### PR DESCRIPTION
**Pull request by Llewellynvdm**
- added site-view switch to allow default public access if access is set on the site view as explained in gh-113

### CODE REVIEW

File: admin/views/admin_views/view.html.php [hunk 1]
```diff
@@ -149,7 +149,12 @@ protected function addToolBar()
 			{
 				JToolBarHelper::custom('admin_views.exportData', 'download', '', 'COM_COMPONENTBUILDER_EXPORT_DATA', true);
 			}
-                } 
+                }
+		if ($this->user->authorise('joomla_component.import_components'))
+		{
+			// add Import Components button.
+			JToolBarHelper::custom('joomla_components.smartImport', 'upload', '', 'COM_COMPONENTBUILDER_IMPORT_COMPONENTS', false);
```
Comment (ro-ot) [inline admin/views/admin_views/view.html.php]: This is a bug, or do you really want the import of components to show in every backend view? I must have missed this before. Sorry!

Comment (Llewellynvdm) [inline admin/views/admin_views/view.html.php]: no you are right... This is a bug!

File: admin/views/joomla_components/view.html.php [hunk 2]
```diff
@@ -164,7 +159,12 @@ protected function addToolBar()
 			{
 				JToolBarHelper::custom('joomla_components.exportData', 'download', '', 'COM_COMPONENTBUILDER_EXPORT_DATA', true);
 			}
-                } 
+                }
+		if ($this->user->authorise('joomla_component.import_components'))
+		{
+			// add Import Components button.
+			JToolBarHelper::custom('joomla_components.smartImport', 'upload', '', 'COM_COMPONENTBUILDER_IMPORT_COMPONENTS', false);
```
Comment (ro-ot) [inline admin/views/joomla_components/view.html.php]: Why did you move this? the buttons are now no longer next to each other.

File: admin/helpers/compiler/e_Interpretation.php [hunk 3]
```diff
@@ -13664,6 +13827,13 @@ public function setAccessSections()
 					$this->langContent['admin'][$siteTitle]	= $siteName.' (Site) Access';
 					$this->langContent['admin'][$siteDesc]	= ' Allows the users in this group to access site '.ComponentbuilderHelper::safeString($siteName,'w').'.';
 					$this->componentGlobal[$sortKey]              = "\t\t".'<action name="site.'.$siteCode.'.access" title="'.$siteTitle.'" description="'.$siteDesc.'" />';
+					
+					// check if this site view requires access rule to default to public
+					if (isset($site_view['public']) && $site_view['public'] == 1)
+					{
+						// we use one as public group (TODO we see if we run into any issues)
+						$this->assetsRules[] = '"site.'.$siteCode.'.access":{"1":1}';
```
Comment (ro-ot) [inline admin/helpers/compiler/e_Interpretation.php]: Nice so we can just extend on this **assetsRules** for the fields and back-ends. So it works just like the old **paramsBuilder** now **extensionsParams**. I think this will work great!

File: admin/views/joomla_components/view.html.php (hunk 2 above)
Comment (Llewellynvdm) [inline admin/views/joomla_components/view.html.php]: Well if you removed all components then you could not import any... so I moved the button to function only that does not requirer a selection, so it will show even if there is no items. I can see it messed up the lineup... :-1:  will see if we can fix that. But it has to show even if there is no items.

File: admin/helpers/compiler/e_Interpretation.php (hunk 3 above)
Comment (Llewellynvdm) [inline admin/helpers/compiler/e_Interpretation.php]: Yea it is a start!

**Comment by Llewellynvdm** [comment]
@mwweb is this working for you? can I bump the version upgrade?

**Comment by mwweb** [comment]
This worked fine on the demo component.

I think this is a good start on it. At least the site views can be automatically set to public. Perhaps sometime in the future we can look at, and discuss, extending this even further. An example, create and edit permissions you generally wouldn't want public, but at least registered. But that can be a start.
